### ISSUE TITLE
How to build to nsp file without out forward nsp app

### ISSUE THREAD

**Issue by PhamMinhKha** [TooTallNate/nx.js#91]
I want to export an nsp file to post to tinfoil Host but don't know how. 

From google translate

**Comment by TooTallNate** [comment]
Can you explain why you don't want to use a forwarder for this case?

**Comment by PhamMinhKha** [comment]
I am trying to help users easily install applications through the NSP format, which will allow them to open the application like a game (because my application requires non-applet mode to decompress zip files), and users can easily install applications through Tinfoil.

**Comment by TooTallNate** [comment]
Ya, makes sense. This is a more general question though, and not strictly related to nx.js. I am also curious if there is a sanctioned method of doing this (it could be useful to add as an option for [NSP Forwarder](https://nsp-forwarder.n8.io/)), but I'll close the issue here for now.

If you do figure it out, report back here and let me know!

**Comment by TooTallNate** [comment]
I'm re-opening this. I have something cooking <picture data-single-emoji=":chef-brb:" title=":chef-brb:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/chef-brb/f115d30d4f321031.png" alt=":chef-brb:" align="absmiddle"></picture>

**Comment by PhamMinhKha** [comment]
> I'm re-opening this. I have something cooking <img alt=":chef-brb:" width="20" height="auto" src="https://camo.githubusercontent.com/69ab6d51f1a9e9ee641dd524bbfb81e876866df1954740fae0573465448c55a7/68747470733a2f2f656d6f6a692e736c61636b2d656467652e636f6d2f5430434151303054552f636865662d6272622f663131356433306434663332313033312e706e67">

Thank you [TooTallNate](https://github.com/TooTallNate)

I just completed the example of the unzip function using nx.js.
https://github.com/PhamMinhKha/nxjs-unzip

**Comment by TooTallNate** [comment]
Ok, I've landed the initial support for NSP files. You'll need [`hacbrewpack`](https://github.com/The-4n/hacBrewPack) installed. You also need to have your `prod.keys` file generated from Lockpick RCM.

Then, you can run the `npm run nsp` script to generate the NSP file (run `npm create nxjs-app@latest` to get the updated dependencies / scripts in `package.json`).

The output is a little verbose at the moment. I might end up hiding the output from `hacbrewpack`, but it looks something like this:

<img width="1552" alt="nsp" src="https://github.com/TooTallNate/nx.js/assets/71256/8d968496-2282-4dc3-a006-ddf97ebd0828">

Let me know if it works for you!

**Comment by PhamMinhKha** [comment]
Wow. You're amazing and you did it very quickly. Everything works fine when I switch to NSP, but the icon doesn't show up.
![z5163975648636_40eae0c9a7117a1d958cf2eae5f75395](https://github.com/TooTallNate/nx.js/assets/12944689/6bb31e1c-a867-4106-a249-07bb80db7e83)

**Comment by PhamMinhKha** [comment]
After trying a few times, I created a folder named 'logo' and renamed icon.jpg to NintendoLogo.png, then the icon reappeared.

**Comment by TooTallNate** [comment]
Ya I was noticing that too. I had a feeling it was because of the missing `logo` files. I'll fix that.

**Comment by TooTallNate** [comment]
I've added some fixes in v0.0.36 and the icon is working properly for me now. Let me know how it goes for you.

**Comment by PhamMinhKha** [comment]
After I updated to version 0.0.36 and deleted the logo folder, the logo displayed properly.

![Home_Menu_20240216_070237_00](https://github.com/TooTallNate/nx.js/assets/12944689/75d03d98-a0ac-4a6c-b765-24460c963584)

**Comment by PhamMinhKha** [comment]
Hi @TooTallNate, I recently updated from version 0.0.36 to 0.0.43 of your package, and encountered an error when running 'npm run nsp'. However, after rolling back to version 0.0.36, everything worked as expected. Could you please assist me with this issue?
```
Running `hacbrewpack`:
  node:internal/modules/cjs/loader:1147
    throw err;
    ^
  
  Error: Cannot find module 'D:\1-code\VietHoa9Tech\node_modules\@nx.js\nsp\hacbrewpack.cjs'
      at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
      at Module._load (node:internal/modules/cjs/loader:985:27)
      at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
      at node:internal/main/run_main_module:28:49 {
    code: 'MODULE_NOT_FOUND',
    requireStack: []
  }
  
  Node.js v20.11.0
```
![image](https://github.com/TooTallNate/nx.js/assets/12944689/2da51936-bfda-44a6-9aa8-a0d149252e20)

**Comment by TooTallNate** [comment]
Thanks for the report. Should be fixed in v0.0.44.
